emcc.py

```diff
@@ -3415,17 +3422,22 @@ def modularize():
   logger.debug('Modularizing, assigning to var ' + shared.Settings.EXPORT_NAME)
   src = open(final).read()
 
+  return_value = shared.Settings.EXPORT_NAME + '.ready'
```

Comment:
Remind me again why couldn't the promise be the returned object itself?

Reply:
[Tried and discussed here](https://github.com/emscripten-core/emscripten/pull/10697#discussion_r413172918)

[Alternative approach suggested and agreed on here](https://github.com/emscripten-core/emscripten/pull/10697#issuecomment-624892484)